File: problems_051to075/problem_057/solution.py
import os
import sys

PROJECT_PATH = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
sys.path.append(PROJECT_PATH)

from measurement import Stopwatch


class Solution(Stopwatch):
    def __init__(self):
        super().__init__()

    def execute(self):
        answer = 0

        numerator = 1
        denominator = 1

        for _ in range(1000):
            numerator, denominator = (numerator + 2*denominator), (numerator + denominator)

            if len(str(numerator)) > len(str(denominator)):
                answer += 1

        return answer

if __name__ == '__main__':
    soln = Solution()
    result = soln.execute()
    print(result)

File: problems_051to075/problem_057/solution.ts
import { SolutionFunction } from '../../lib/types';

const solution: SolutionFunction = () => {
  let answer: ReturnType<SolutionFunction> = 0;

  let numerator = 1n;
  let denominator = 1n;

  for (let num = 0; num < 1000; num += 1) {
    [numerator, denominator] = [(numerator + 2n * denominator), (numerator + denominator)];

    if (String(numerator).length > String(denominator).length) {
      answer += 1;
    }
  }

  return answer;
};

export default solution;
